### PR TITLE
plover support in cmake, demo program

### PR DESCRIPTION
**Pull request by kovach**
don't want to merge yet; adds cabal/ghc dependency to build process

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,6 +65,7 @@ add_subdirectory(clapack-3.2.1-CMAKE)
 add_subdirectory(CBLAS)
 add_subdirectory(lapacke)
 add_subdirectory(src)
+add_subdirectory(plover)
```
Comment (fnoble) [inline CMakeLists.txt]: maybe plover should be a subdirectory of src, seeing as it contains source code? but don't feel strongly about this, also see the benefit of keeping it separate

File: plover/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+add_custom_command (
+  OUTPUT "${PROJECT_SOURCE_DIR}/plover/plover.c" "${PROJECT_SOURCE_DIR}/plover/plover.h"
```
Comment (fnoble) [inline plover/CMakeLists.txt]: maybe the output at least should be moved into `src/`? or at least out the way of the haskell source?

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -38,13 +40,18 @@ set(libswiftnav_SRCS
...
+SET_SOURCE_FILES_PROPERTIES("${PROJECT_SOURCE_DIR}/plover/plover.c" PROPERTIES GENERATED TRUE)
+
+add_library(swiftnav-static STATIC ${libswiftnav_SRCS} ${libswiftnav_GENERATED_SRCS})
 target_link_libraries(swiftnav-static cblas)
 target_link_libraries(swiftnav-static lapacke)
 install(TARGETS swiftnav-static DESTINATION lib${LIB_SUFFIX})
 
 if(BUILD_SHARED_LIBS)
-  add_library(swiftnav SHARED ${libswiftnav_SRCS})
+  add_library(swiftnav STATIC ${libswiftnav_SRCS} ${libswiftnav_GENERATED_SRCS})
```
Comment (fnoble) [inline src/CMakeLists.txt]: did you mean to change `SHARED` -> `STATIC` here?

**Comment by fnoble** [comment]
looking good!

Maybe change the target name to `plover` or `generate` or something so that you can do `make generate` to regenerate?

**Comment by fnoble** [comment]
Changes required before merging:

 - [ ] plover code only gets built if ghc/plover is available 
 - [ ] only rebuild if the source files are newer than the generated output
 - [ ] finalize directory structure

**Comment by kovach** [comment]
closed in favor of #120

**Pull request closed without merging**